src/Middleware/InjectDetector.php

```diff
@@ -28,12 +28,11 @@ public function handle(Request $request, \Closure $next)
             // no-op.
         }
 
-        if (! $response instanceof Response) {
-            return $next($request);
+        // Inject the detector when returning an HTTP response
```

Comment:
This condition is the same as before, it's just not an early return.

Reply:
The difference here is that if we have an `Illuminate\Http\JsonResponse` instead of an `Illuminate\Http\Response` it will pass through without having the detector injected.  It is my understanding that the `JsonResponse` class does not extend the regular `Response` class, so it will fail this check.

Reply:
I suppose it might make more sense to make this decision based on whether the request is expecting JSON - I will update the PR.

Reply:
Never mind - it looks like Livewire requests will accept either JSON or HTML, so this may already be the best method to determine if the application is returning JSON.